partiql-planner/src/test/kotlin/org/partiql/planner/internal/typer/PTypeMetaInPlan.kt

```diff
@@ -0,0 +1,148 @@
+package org.partiql.planner.internal.typer
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.partiql.ast.Ast.exprCast
+import org.partiql.ast.Ast.exprLit
+import org.partiql.ast.Ast.query
+import org.partiql.ast.DataType
+import org.partiql.ast.Literal
+import org.partiql.ast.Statement
+import org.partiql.plan.Action
+import org.partiql.plan.rex.RexCast
+import org.partiql.planner.PartiQLPlanner
+import org.partiql.spi.catalog.Session
+import org.partiql.spi.types.PType
+
+private const val UNSPECIFIED_LENGTH = "UNSPECIFIED_LENGTH"
+private const val UNSPECIFIED_PRECISION = "UNSPECIFIED_PRECISION"
+private const val UNSPECIFIED_SCALE = "UNSPECIFIED_SCALE"
+
+/**
+ * Tests that PType metas are available in the plan after AST -> plan conversion and type inference.
+ */
+class PTypeMetaInPlan {
+    private val planner = PartiQLPlanner.standard()
+    private fun PType.assertHasLengthMeta() {
+        assertEquals(this.metas[UNSPECIFIED_LENGTH], true)
+    }
+
+    private fun PType.assertHasPrecisionMeta() {
+        assertEquals(this.metas[UNSPECIFIED_PRECISION], true)
+    }
+
+    private fun PType.assertHasScaleMeta() {
+        assertEquals(this.metas[UNSPECIFIED_SCALE], true)
+    }
```

Comment:
The `assertHas*` names are minorly confusing. Maybe replace with `assertUnspecified*`?